### ISSUE TITLE
chore: extract paired_wfcv scaffolding into experiments/lib/

### ISSUE THREAD

**Issue by edwinsteele** [edwinsteele/fuel-price-signal#229]
## What

Extract the duplicated paired-walk-forward-CV scaffolding from the dated experiment scripts into a shared `experiments/lib/` package, so future experiment scripts contain candidate-feature logic + run grid + cohort masks and almost nothing else. Refactor only; no behaviour change.

Proposed package shape:

```
experiments/lib/
  __init__.py
  constants.py    # SEEDS=(42..46), SHOCK_FOLDS={1,4,9,13}, FOLD_PARAMS, LGBM_DEFAULTS
  fit.py          # fit_score(train_df, val_df, cols, seed) -> (ll, p, t)
                  # per_row_log_loss(y, p) -> np.ndarray
  folds.py        # iter_folds_with_baseline_fit(df, baseline_cols, seed) ->
                  #   yields (fold_idx, regime, train_df, val_df,
                  #           baseline_ll, baseline_p, baseline_t, baseline_prl)
                  # Encapsulates the "fit baseline once per fold, reuse for R0+seed0" trick.
                  # Does NOT take cohort masks — caller builds those from baseline_prl.
  cohorts.py      # hard_quantile_mask(prl, q) -> bool mask
  gates.py        # seed_variance_gate(df_rows, cohort_ll_map) -> (summary, flags)
                  #   prints flagged cells; raises on degenerate cohort_median (NaN/<=0)
  aggregate.py    # aggregate_with_deltas(df_rows, cohort_ll_map, baseline_run="R0")
                  #   -> fold_run DataFrame with *_mean, *_median, *_seedstd, delta_*
                  # fold_aggregate(df, col, mask=None) -> dict
                  #   (single-seed reducer from phase_lookup/step4_paired_wfcv)
                  # named_regressions(df_rows, col, threshold) -> prints fold regressions
                  #   (lifted from phase_lookup/step4; missing from newer scripts)
  io.py           # to_jsonable(o); write_meta(out_dir, meta)
  timing.py       # @contextmanager time_block(label) — prints "[label] N.Ns"
  README.md       # one-paragraph-per-module reference
```

API constraints derived from the existing scripts:

- The per-fold loop stays IN THE EXPERIMENT SCRIPT. `iter_folds_with_baseline_fit` only yields baseline outputs; the script builds its own cohort masks (`hard25`, `hard10`, `lated`, `ext_descent_shallow`, etc) on top. Hiding the per-fold loop would push thinking out of view of the reader.
- Multi-seed gate + median aggregator and single-seed `fold_aggregate` coexist in `aggregate.py`. Do not try to unify them.
- LightGBM params (`verbose=-1, subsample=0.8, subsample_freq=1`) move into `LGBM_DEFAULTS`; `fit_score` reads from there. Drift between scripts on these params has cost a PR already.
- Constants drift is exactly the thing this PR exists to prevent — `SEEDS` and `SHOCK_FOLDS` MUST be imported, not redefined per script.

## Why I noticed this

Audit of 6 paired_wfcv scripts in `experiments/2026-06-0{5,6,7,8,9}_*/` confirmed each is ~400-470 lines, ~160 of which are near-verbatim duplicated scaffolding (`fit_score`, `per_row_log_loss`, seed-variance gate ~40 lines, aggregation+delta ~25 lines, `_to_jsonable`, walltime instrumentation, per-fold print header). Multi-step experiments multiply this — `2026-06-06_late_descent_triplet/` has 10+ scripts, `2026-06-05_phase_lookup_nonparametric/` has 6, each regenerating the scaffolding.

Beyond raw line count, the duplication is **drift-prone**: the seed-variance gate's "5×" vs "5x" wording diverges between scripts; the named-regression printer exists in `phase_lookup/step4` but was dropped from newer scripts; LGBM params have drifted before.

This is foundational scaffolding for the next several experiments. The line saving is real but the bigger lever is **turn-count reduction when drafting a new experiment script**: an Opus session writing "discuss candidate features → script" currently regenerates ~160 lines of boilerplate it has already produced for prior experiments. With this lib, the first draft is mostly the candidate-feature function + RUNS dict + per-experiment cohort masks.

## Out of scope (file as Phase 2)

- Feature-computation primitives (`cohort_std_by_date`, `calendar_aware_delta`, `rolling_baseline_per_date`, `px_change_lag_diagnostic`). These recur across 4+ scripts and are higher leverage than the orchestration helpers, but have more API-shape questions and warrant their own design issue. File once this lands.
- Per-fold refit pattern from `phase_lookup_nonparametric/step4` (`fit_lookup` / `apply_lookup`). Only one use case today; formalize when a second appears.
- The older single-window style in `2026-06-03_drop_redundant_pair/run.py` is superseded and should NOT be brought under the lib.

## Files likely affected

- `experiments/lib/` (new package, modules listed above)
- `experiments/lib/README.md` (new)
- `experiments/2026-06-09_shallow_elongated/paired_wfcv.py` — retrofit for bit-equivalence validation. Pick this one because it has both `rowpreds.parquet` dump and the standard cohorts; if the retrofit works there, it works everywhere.
- `docs/CONVENTIONS.md` — add new `## Experiment scripts` section pointing at `experiments/lib/`; cross-link to the existing `feedback_load_features_helper`, `feedback_experiment_scripts_pythonpath`, `feedback_instrument_walltime`, `feedback_throwaway_validation_scripts` rules so the experiment-scripts conventions are findable in one place.
- Memory: new entry `feedback_experiments_lib.md` saying "new paired_wfcv scripts use `experiments/lib/` helpers; do not re-implement scaffolding inline." Index it in `MEMORY.md` adjacent to the other experiment-scripts feedback entries.
- `AGENTS.md` — add a short reference to `experiments/lib/` under whichever section covers experiments (if applicable).

## Acceptance criteria

- [ ] `experiments/lib/` package exists with the 8 modules above + `README.md`.
- [ ] `experiments/2026-06-09_shallow_elongated/paired_wfcv.py` is retrofitted to use the lib.
- [ ] Retrofit produces **bit-identical** `runs.csv`, `fold_run.csv`, and `meta.json` (modulo `total_wall_seconds` / per-step `*_s` timing fields which are non-deterministic by nature). Compare via `git diff --no-index` on the pre/post outputs; include the diff in PR description.
- [ ] The retrofit script is ≤ 250 lines (down from 447) — this is a soft check that the extraction actually pulled enough out.
- [ ] `docs/CONVENTIONS.md` has a new `## Experiment scripts` section that:
  - States: new paired_wfcv scripts MUST use `experiments/lib/` helpers; do not copy scaffolding from prior scripts.
  - Lists each lib module + one-line purpose.
  - Cross-references the existing experiment-script rules (load_features, PYTHONPATH=., wall-time instrumentation, throwaway style for one-off validation).
- [ ] Memory entry `feedback_experiments_lib.md` filed and indexed in `MEMORY.md`.
- [ ] `uv run ruff check . && uv run pytest -q` clean.
- [ ] No changes to `fuel_signal/**`, `data/**`, `experiments/results.csv`, or scripts under `experiments/2026-06-03_*` (superseded style) or `experiments/cv_compare_*`, `experiments/redundancy_*`, `experiments/shap*` (out of scope older diagnostics).

**Comment by edwinsteele** [comment]
Relabelled `chore` → `design`.

Reason: acceptance criteria require a bit-equivalence diff against the retrofitted `shallow_elongated/paired_wfcv.py`, which means running the script (~210 LightGBM fits on `data/features.csv`). The hourly worker routine has an empty dev DB and per the project's no-experiment-run rule does not execute training scripts, so it cannot produce or verify the diff. Issue is interactive-session work end-to-end.

If the lib extraction is split from the retrofit validation later, the extraction half could become worker-doable; for now keeping them together since the retrofit is what validates the API shape.
